backend/controllers/OrderController.js

```diff
@@ -58,28 +55,44 @@ class OrderController {
    * @param {Object} req - Express request object containing order data.
    * @param {Object} res - Express response object.
    */
-async addOrder(req, res) {
-  try {
-    console.log("Request body:", req.body); // Log the incoming request body
-    const userId = req.user?.id || "test-user-id"; // Default userId for testing
+  async addOrder(req, res) {
+    try {
+      const userId = req.user?.id || "test-user-id";
+      const { orders } = req.body; // array of { productId, price, description, quantity }
 
-    const { orders } = req.body; // Expecting an array of order items
-    if (!orders || !Array.isArray(orders) || orders.length === 0) {
-      return res.status(400).json({ error: "Invalid or missing order data." });
-    }
+      if (!orders || !Array.isArray(orders) || orders.length === 0) {
+        return res.status(400).json({ error: "Invalid or missing order data." });
+      }
 
-    const orderPromises = orders.map(({ productId, price, description, quantity }) => {
-      return OrderModel.addOrderManually(productId, price, description, quantity, userId);
-    });
+      const orderPromises = orders.map(({ productId, price, description, quantity }) => {
+        return OrderModel.addOrderManually(productId, price, description, quantity, userId);
+      });
 
-    const newOrders = await Promise.all(orderPromises);
-    res.status(201).json(newOrders);
-  } catch (error) {
-    console.error("Error in addOrder:", error);
-    res.status(500).json({ error: "Failed to add order." });
+      const newOrders = await Promise.all(orderPromises);
+      res.status(201).json(newOrders);
+    } catch (error) {
+      console.error("Error in addOrder:", error);
+      res.status(500).json({ error: "Failed to add order." });
+    }
   }
-}
 
+    /**
+   * Cancel (delete) an existing order.
+   * @param {Object} req - Express request object containing orderId in `req.params.id`.
+   * @param {Object} res - Express response object.
+   */
+  async cancelOrder(req, res) {
```

Comment:
I can't find a way to cancel an order in the app, is this currently being used?

Reply:
No, it was for Yukun to use on order history, but we don't have to use it.